### ISSUE TITLE
Pomodoro-like behavior for mi band

### ISSUE THREAD

**Issue by aafa** [Freeyourgadget/Gadgetbridge#151]
I'd love to see my mi band doing time tracking in Pomodoro-like style (http://pomodorotechnique.com/)

Basic idea
1. In the beginning of working period mi band vibrates showing 1 red led
2. As times goes by it shows 2 then 3 leds 
3. At the beginning of resting period it vibrates and shows 1 green led
4. Then 2-3 leds, finally going back to step 1.

AFAIK we can't have this whole thing triggered from band itself only from mobile. So things like leds showing can be tricked in a way as described here (https://github.com/Freeyourgadget/Gadgetbridge/issues/80#issuecomment-126023503)
And vibration can be simply triggered from mobile, right?

Concerns
1. Out of range from mobile it wont work (or some part of it will?)
2. Battery consumption
3. Leds are tricky

Any thoughts are appreciated - how possible is this, what pitfalls?

**Comment by cpfeiffer** [comment]
Interesting idea. Generalizing this, it looks like we need some kind of API, scripting interface to notifications (what you describe is basically some very custom notification).

My only concerns are:
- newer Mi Bands do not have coloured LEDs anymore
- old Mi Bands with newer firmware do not allow fine grained LED notifications anymore (AFAIU)
- even with old Mi Band and "old" firmware, you cannot control LEDs separately, but as suggested, one could use a different lighting scheme for that

Battery consumption should not be a concern. Or do you mean that the LEDs should light *all the time*?

**Comment by aafa** [comment]
> Or do you mean that the LEDs should light all the time?

No, I see it working like step plan progress. Plus some additional gesture - like tapping - to see current stage.  
I mentioned battery in respect of that trickiness needed to simulate leds going on\off by modifying parameters of band from mobile often then usual.

**Comment by cpfeiffer** [comment]
Handling feedback from the band is indeed not easily possible, see #63 What could be done is enable realtime sensor information, detect certain "gestures" on the phone and react to that. However that would probably indeed be battery consuming.

Sending notifications in different colours should not be a problem at all.

**Comment by sarg** [comment]
I think, that tasker support would be sufficient to fulfill this and many other interesting ideas.

**Comment by cpfeiffer** [comment]
Since tasker is not OSS, we should [additionally] provide something free, even if it's just some kind of import of a textual script.

**Comment by dj0001** [comment]
you could use a ([webapp](https://dj0001.github.io/miband3-selfie/notify.htm)) countdown-timer with notifications. Set display sleep to never (because of missing [WakeLock](https://developers.google.com/web/updates/2018/12/wakelock))  
Allow notification and possibly change timer.
